zeroex/orderwatch/order_watcher.go

```diff
@@ -1158,89 +1158,53 @@ func (w *Watcher) convertValidationResultsIntoOrderEvents(
-						SignedOrder:              order.SignedOrder(),
-						FillableTakerAssetAmount: order.FillableTakerAssetAmount,
-						EndState:                 zeroex.ESOrderUnexpired,
-					}
-					orderEvents = append(orderEvents, orderEvent)
-				}
-				// No important state-change happened. Still want to update lastValidatedBlock
-				w.updateOrderLastValidatedBlock(order, validationBlock)
-				continue
-			}
-			if oldFillableAmount.Cmp(big.NewInt(0)) == 1 && oldAmountIsMoreThenNewAmount {
-				// If order was previously expired, check if it has become unexpired
-				if order.IsRemoved && oldFillableAmount.Cmp(big.NewInt(0)) != 0 && !isOrderExpired {
-					w.rewatchOrder(order, newFillableAmount, validationBlock)
-					orderEvent := &zeroex.OrderEvent{
-						Timestamp:                validationBlock.Timestamp,
-						OrderHash:                order.Hash,
-						SignedOrder:              order.SignedOrder(),
-						FillableTakerAssetAmount: order.FillableTakerAssetAmount,
-						EndState:                 zeroex.ESOrderUnexpired,
-					}
-					orderEvents = append(orderEvents, orderEvent)
-				} else {
-					w.updateOrderFillableTakerAssetAmount(order, newFillableAmount, validationBlock)
-				}
-				// Order was filled, emit event
+			if isOrderUnexpired {
+				w.rewatchOrder(order, newFillableAmount, validationBlock)
 				orderEvent := &zeroex.OrderEvent{
 					Timestamp:                validationBlock.Timestamp,
-					OrderHash:                acceptedOrderInfo.OrderHash,
+					OrderHash:                order.Hash,
 					SignedOrder:              order.SignedOrder(),
-					EndState:                 zeroex.ESOrderFilled,
-					FillableTakerAssetAmount: acceptedOrderInfo.FillableTakerAssetAmount,
-					ContractEvents:           orderHashToEvents[order.Hash],
+					FillableTakerAssetAmount: order.FillableTakerAssetAmount,
+					EndState:                 zeroex.ESOrderUnexpired,
 				}
 				orderEvents = append(orderEvents, orderEvent)
-			} else if oldFillableAmount.Cmp(big.NewInt(0)) == 1 && !oldAmountIsMoreThenNewAmount {
-				// The order is now fillable for more then it was before. E.g.: A fill txn reverted (block-reorg)
-				// If order was previously expired, check if it has become unexpired
-				if order.IsRemoved && oldFillableAmount.Cmp(big.NewInt(0)) != 0 && !isOrderExpired {
-					w.rewatchOrder(order, newFillableAmount, validationBlock)
-					orderEvent := &zeroex.OrderEvent{
-						Timestamp:                validationBlock.Timestamp,
-						OrderHash:                order.Hash,
-						SignedOrder:              order.SignedOrder(),
-						FillableTakerAssetAmount: order.FillableTakerAssetAmount,
-						EndState:                 zeroex.ESOrderUnexpired,
-					}
-					orderEvents = append(orderEvents, orderEvent)
-				} else {
-					w.updateOrderFillableTakerAssetAmount(order, newFillableAmount, validationBlock)
+			} else {
+				w.updateOrderFillableTakerAssetAmount(order, newFillableAmount, validationBlock)
```

Comment:
I renamed it to updateOrderFillableTakerAssetAmountAndBlockInfo. I'm going to make another PR that unifies `updateOrderFillableTakerAssetAmountAndBlockInfo`, `rewatchOrder`, and `unwatchOrder`, since they're all the same, except for how they handle `IsRemoved`.